koji-setup/mash.sh

```diff
@@ -16,30 +13,95 @@ MASH_TRACKER_DIR="$MASH_DIR"/latest
 MASH_DIR_NEW="$MASH_TRACKER_DIR".new
 
+create_dist_repos() {
+	local output_dir="${1}"
+
+	local work_dir="$(mktemp -d)"
+
+	local nvr_pkg_list="${work_dir}/nvr-pkg-list"
+	local bin_rpm_paths="${work_dir}/bin-rpm-paths"
+	local debuginfo_rpm_paths="${work_dir}/debuginfo-rpm-paths"
+	local src_rpm_paths="${work_dir}/src-rpm-paths"
+	local comps_file="${work_dir}/comps.xml"
+
+	sed -r -e 's|[^/]+/||' -e "s|^|${KOJI_DIR}/|" "${KOJI_REPO_PATH}/${BUILD_ARCH}/pkglist" > "${bin_rpm_paths}"
+	sed -r -e 's|^([^/]+/){2}||' -e 's|/[^/]+/[^/]+$||' "${KOJI_REPO_PATH}/${BUILD_ARCH}/pkglist" | sort -u > "${nvr_pkg_list}"
+	while IFS='/' read -r name version release; do
+		local debuginfo_rpm_path="${KOJI_DIR}/packages/${name}/${version}/${release}/${BUILD_ARCH}/${name}-debuginfo-${version}-${release}.${BUILD_ARCH}.rpm"
+		if [[ -s "${debuginfo_rpm_path}" ]]; then
+			echo "${debuginfo_rpm_path}" >> "${debuginfo_rpm_paths}"
+		fi
+		echo "${KOJI_DIR}/packages/${name}/${version}/${release}/src/${name}-${version}-${release}.src.rpm" >> "${src_rpm_paths}"
+	done < "${nvr_pkg_list}"
+
+	cp -f "${KOJI_REPO_PATH}/groups/comps.xml" "${comps_file}"
+
+	make_repo "${output_dir}/clear/${BUILD_ARCH}/os" "Packages" "${bin_rpm_paths}" "${comps_file}" &
+	make_repo "${output_dir}/clear/${BUILD_ARCH}/debug" "." "${debuginfo_rpm_paths}" &
+	make_repo "${output_dir}/clear/source/SRPMS" "." "${src_rpm_paths}" &
+	wait
+
+	create_dnf_conf "${work_dir}/dnf-os.conf" "${output_dir}/clear/${BUILD_ARCH}/os" clear-os
+	create_dnf_conf "${work_dir}/dnf-debug.conf" "${output_dir}/clear/${BUILD_ARCH}/debug" clear-debug
+	create_dnf_conf "${work_dir}/dnf-SRPMS.conf" "${output_dir}/clear/source/SRPMS" clear-SRPMS
+
+	write_packages_file "${work_dir}/dnf-os.conf" "$output_dir/clear/$BUILD_ARCH/packages-os"
+	write_packages_file "${work_dir}/dnf-debug.conf" "$output_dir/clear/$BUILD_ARCH/packages-debug"
+	write_packages_file "${work_dir}/dnf-SRPMS.conf" "$output_dir/clear/source/packages-SRPMS"
+
+	rm -rf "${work_dir}"
+}
+
+make_repo() {
+	local repo_dir="${1}"
+	local rpm_dir="${1}/${2}"
+	local file_list="${3}"
+	local comps_file="${4}"
+
+	local create_repo_cmd="createrepo_c --quiet --database --compress-type xz --workers $(nproc --all)"
+
+	mkdir -p "${rpm_dir}"
+	xargs -a "${file_list}" -I {} cp -f {} "${rpm_dir}"
+	mkdir -p "${repo_dir}"
+	if [[ -z "${comps_file}" ]]; then
+		${create_repo_cmd} "${repo_dir}"
+	else
+		${create_repo_cmd} --groupfile "${comps_file}" "${repo_dir}"
+	fi
+}
+
+create_dnf_conf() {
+	local dnf_conf="${1}"
+	local repo_path="${2}"
+	local repo_name="${3:-clear}"
+	cat > "${dnf_conf}" <<EOF
+[${repo_name}]
+name=${repo_name}
+baseurl=file://${repo_path}
+EOF
+}
+
 write_packages_file() {
-	local PKG_DIR="$1"
-	local PKG_FILE="$2"
-	rpm -qp --qf="%{NAME}\t%{VERSION}\t%{RELEASE}\n" "$PKG_DIR"/*.rpm | sort > "$PKG_FILE"
+	local dnf_conf="${1}"
+	local output_file="${2}"
+
+	dnf --config "${dnf_conf}" --quiet --releasever=clear \
```

Comment:
Why are you switching from rpm to dnf here?

Reply:
Because package lists do not generate anymore for the os repo, but still do for the SRPMS and debug repos.  We're probably hitting a character limit here.